### PR TITLE
Sub: Issue 9801. Get Rid of Step Input on Engaging Input_Hold

### LINKED ISSUE #9801: Sub: There is a step input when setting input hold
Moved from: bluerobotics/ardusub#165
Original author: @jaxxzer

### Feature request
In the moment that input hold is engaged, any stick displacement from center is additive to where the input hold target is. That means that if the stick is displaced and input hold is subsequently engaged, the input instantly doubles until the stick returns to center (and the desired offset is then obtained).

**Platform**
[ ] All
[ ] AntennaTracker
[ ] Copter
[ ] Plane
[ ] Rover
[x] Submarine

### PR DESCRIPTION
**Pull request by jaylthegreat**
Closes #9801 

Tested in SITL.

### CODE REVIEW

File: ArduSub/joystick.cpp [hunk 1]
```diff
@@ -335,11 +348,12 @@ void Sub::handle_jsbutton_press(uint8_t button, bool shift, bool held)
             yTrim = abs(y_last) > 50 ? y_last : 0;
             bool input_hold_engaged_last = input_hold_engaged;
             input_hold_engaged = zTrim || xTrim || yTrim;
-            if (input_hold_engaged) {
+            if (input_hold_engaged) {
```
Comment (amilcarlucas) [inline ArduSub/joystick.cpp]: whitespace change

File: ArduSub/joystick.cpp [hunk 2]
```diff
@@ -93,17 +94,29 @@ void Sub::transform_manual_control_to_rc_override(int16_t x, int16_t y, int16_t
...
+    int16_t zTot = z + zTrim;
+    int16_t yTot = y + yTrim;
+    int16_t xTot = x + xTrim;
+
+    if (!controls_reset_since_input_hold) {
+        zTot = zTrim + 500; // 500 is neutral for throttle
+        yTot = yTrim;
+        xTot = xTrim;
+        // if all 3 axes return to neutral, than we're ready to accept input again
+        controls_reset_since_input_hold = (z == 500 && y == 0 && x == 0);
```
Comment (jaxxzer) [inline ArduSub/joystick.cpp]: exactly equal to zero on all channels is a bit restrictive. I think we should expand this to a +/-5% or so range.

File: ArduSub/joystick.cpp [hunk 3]
```diff
@@ -93,17 +94,29 @@ void Sub::transform_manual_control_to_rc_override(int16_t x, int16_t y, int16_t
...
+    }
```
Comment (jaxxzer) [inline ArduSub/joystick.cpp]: The assignments in lines 97~99 should be placed in an `else` here.

**Comment by Williangalvani** [comment]
Just tested in the ROV. It is working fine.

**Pull request merged**